File: app/src/main/java/ch/unstable/ost/preference/SettingsActivity.java
package ch.unstable.ost.preference;

import android.os.Bundle;
import android.preference.PreferenceActivity;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatDelegate;

import java.util.List;

import ch.unstable.ost.R;
import ch.unstable.ost.theme.ThemeHelper;

public class SettingsActivity extends PreferenceActivity implements AppearanceSettingsFragment.OnThemeChangedListener {

    static {
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        ThemeHelper.setTheme(this);
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onBuildHeaders(List<Header> target) {
        loadHeadersFromResource(R.xml.pref_headers, target);
    }

    @Override
    public void onThemeChanged() {
        recreate();
    }


    @Override
    protected boolean isValidFragment(String fragmentName) {
        return AppearanceSettingsFragment.class.getName().equals(fragmentName)
                || ContentSettingsFragment.class.getName().equals(fragmentName);
    }
}

File: app/src/main/java/ch/unstable/ost/preference/SettingsActivity.kt
package ch.unstable.ost.preference

import android.os.Bundle
import android.preference.PreferenceActivity
import androidx.appcompat.app.AppCompatDelegate
import ch.unstable.ost.R
import ch.unstable.ost.preference.AppearanceSettingsFragment.OnThemeChangedListener
import ch.unstable.ost.theme.ThemeHelper

class SettingsActivity : PreferenceActivity(), OnThemeChangedListener {
    companion object {
        init {
            AppCompatDelegate.setCompatVectorFromResourcesEnabled(true)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        ThemeHelper.setTheme(this)
        super.onCreate(savedInstanceState)
    }

    override fun onBuildHeaders(target: List<Header>) {
        loadHeadersFromResource(R.xml.pref_headers, target)
    }

    override fun onThemeChanged() {
        recreate()
    }

    override fun isValidFragment(fragmentName: String): Boolean {
        return AppearanceSettingsFragment::class.java.name == fragmentName || ContentSettingsFragment::class.java.name == fragmentName
    }
}